hail/python/hail/ggplot/scale.py

```diff
@@ -217,26 +241,71 @@ def transform(df):
 
 
 class ScaleColorHue(ScaleDiscrete):
-    def create_local_transformer(self, groups_of_dfs):
-        categorical_strings = set()
-        for group_of_dfs in groups_of_dfs:
-            for df in group_of_dfs:
-                if self.aesthetic_name in df.attrs:
-                    categorical_strings.add(df.attrs[self.aesthetic_name])
-
-        num_categories = len(categorical_strings)
+    def get_values(self, categories):
+        num_categories = len(categories)
         step = 1.0 / num_categories
         interpolation_values = [step * i for i in range(num_categories)]
         hsv_scale = px.colors.get_colorscale("HSV")
-        colors = px.colors.sample_colorscale(hsv_scale, interpolation_values)
-        unique_color_mapping = dict(zip(categorical_strings, colors))
-
-        def transform(df):
-            df.attrs[f"{self.aesthetic_name}_legend"] = df.attrs[self.aesthetic_name]
-            df.attrs[self.aesthetic_name] = unique_color_mapping[df.attrs[self.aesthetic_name]]
-            return df
-
-        return transform
+        return px.colors.sample_colorscale(hsv_scale, interpolation_values)
+
+
+class ScaleShapeAuto(ScaleDiscrete):
+    def get_values(self, categories):
+        return [
```

Comment:
These are hardcoded here because there aren't equivalent methods to `plotly.express.colors.(get|sample)_colorscale` for the predefined list of shapes that plotly supports, and I couldn't find somewhere in the plotly API from which to directly import that list of constants.